src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesV2.scala

```diff
@@ -0,0 +1,126 @@
+package org.broadinstitute.dsde.workbench.sam.api
+
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.model.StatusCodes.{NotFound, OK}
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.{Directive0, ExceptionHandler, Route}
+import cats.effect.IO
+import org.broadinstitute.dsde.workbench.model._
+import org.broadinstitute.dsde.workbench.sam.model.api.SamUserResponse._
+import org.broadinstitute.dsde.workbench.sam.model.api.{SamUser, SamUserResponse}
+import org.broadinstitute.dsde.workbench.sam.service.UserService
+import org.broadinstitute.dsde.workbench.sam.util.SamRequestContext
+
+/** Created by tlangs on 10/12/2023.
+  */
+trait UserRoutesV2 extends SamUserDirectives with SamRequestContextDirectives {
+  val userService: UserService
+
+  /** Changes a 403 error to a 404 error. Used when `UserInfoDirectives` throws a 403 in the case where a user is not found. In most routes that is appropriate
+    * but in the user routes it should be a 404.
+    */
+  private val changeForbiddenToNotFound: Directive0 = {
+    import org.broadinstitute.dsde.workbench.model.ErrorReportJsonSupport._
+
+    handleExceptions(ExceptionHandler {
+      case withErrorReport: WorkbenchExceptionWithErrorReport if withErrorReport.errorReport.statusCode.contains(StatusCodes.Forbidden) =>
+        complete((StatusCodes.NotFound, withErrorReport.errorReport.copy(statusCode = Option(StatusCodes.NotFound))))
+    })
+  }
+
+  // These routes are wrapped in `withUserAllowInactive` because a user should be able to get info about themselves
+  // Routes that need the user to be active should be wrapped in a directive, such as `withActiveUser`, to ensure
+  // that the user is allowed to use the system.
+  def userRoutesV2(samRequestContextWithoutUser: SamRequestContext): Route =
+    withUserAllowInactive(samRequestContextWithoutUser) { samUser: SamUser =>
+      val samRequestContext = samRequestContextWithoutUser.copy(samUser = Some(samUser))
+      pathPrefix("users") {
+        pathPrefix("v2") {
+          pathPrefix("self") {
+            // api/users/v2/self
```

Comment:
Hahaha ❤️ I was literally just doing this 5 minutes ago in my own PR so I could understand what the hell the routes are.  This syntax is brutal.

Reply:
Its sooooo bad